src/gui/properties/pieceavailabilitybar.h

```diff
@@ -31,52 +31,33 @@
 #ifndef PIECEAVAILABILITYBAR_H
 #define PIECEAVAILABILITYBAR_H
 
-#include <QWidget>
-#include <QPainter>
-#include <QImage>
+#include "piecesbar.h"
 
-class PieceAvailabilityBar: public QWidget
+class PieceAvailabilityBar: public PiecesBar
 {
+    using base = PiecesBar;
     Q_OBJECT
     Q_DISABLE_COPY(PieceAvailabilityBar)
 
-private:
-    QImage m_image;
+public:
+    PieceAvailabilityBar(QWidget *parent);
 
-    // I used values, because it should be possible to change colors in runtime
+    void setAvailability(const QVector<int> &avail);
 
-    // background color
-    int m_bgColor;
-    // border color
-    int m_borderColor;
-    // complete piece color
-    int m_pieceColor;
-    // buffered 256 levels gradient from bg_color to piece_color
-    QVector<int> m_pieceColors;
+    // PiecesBar interface
+    void clear() override;
+
+private:
+    virtual bool updateImage(QImage &image) override;
```

Comment:
I think it is better to leave it. If one is reading he should know immediately that it is virtual.